### DISCUSSION TITLE
satisfies ChartConfig

### DISCUSSION THREAD

**Discussion by Disrupt3d** [shadcn-ui/ui#4242] [General]
hi,  we  are currently tryng to use  shadcn/charts  on  a  non-typescrit projects ( just nextJS) and  we have some troubles when  tryng to display  charts :   
here is a example : 
const chartConfig = {
  desktop: {
    label: "Desktop",
    color: "#2563eb",
  },
  mobile: {
    label: "Mobile",
    color: "#60a5fa",
  },
} satisfies ChartConfig  

 the satisfies Chartconfig is marked down as an error in our IDE . It seems to be adapted to typescript.
if we remove this,  charts does not  show .
How can we fix this ?

**Comment by shadcn** [comment]
It should work without `ChartConfig` in non-TypeScript project. Can you paste in the chart code here?

**Reply by pablo-chocobar** [reply]
Hi, I ran into the same problem. 

```js
export function RadarChartComponent(props) {
      const chartConfig = {
    stat: {
      label: props.label,
      color: "#e6d5bc",
    }}
    const chartData = props.data;

  return (
    <Card>
      <CardHeader className="items-center pb-4 ">
        <CardTitle>Overview</CardTitle>
        <CardDescription>
        </CardDescription>
      </CardHeader>
      <CardContent className="pb-0">
        <ChartContainer
          config={chartConfig}
          className="mx-auto max-h-[250px]"
        >
          <RadarChart data={chartData}>
            <ChartTooltip cursor={false} content={<ChartTooltipContent />} />
            <PolarAngleAxis dataKey= {props.dataKey} />
            <PolarGrid />
            <Radar
              dataKey= {props.label}
              fill="#957d67"
              fillOpacity={0.6}
            />
          </RadarChart>
        </ChartContainer>
      </CardContent>
    </Card>
  )
} 
```
this is the code im using right now. 

And this is my output:

<img src="https://github.com/user-attachments/assets/ef92bea5-fe98-40fa-a798-783dbc6cd45c" width="320">

Light mode for good measure. 

<img src= "https://github.com/user-attachments/assets/ee337697-cb94-4d5a-8099-5eeda6e56ad0"  width="320">

However i can add Tailwind classes to the Card and ChartContainer to change the background. It'd be great if you can suggest a way to make use of the chartConfig or even just change the color of the grid lines. 

P.S thank you for the work you've done :)

**Comment by hanksgiving** [comment]
Remove the "satisfies ChartConfig" as well as the import of ChartConfig component from @/components/ui/chart

ie.
import {
  <del>ChartConfig,</del>
  ChartContainer
  ChartTooltip,
  ChartTooltipContent,
} from "@/components/ui/chart"

**Comment by Disrupt3d** [comment]
hello guys, thank you for your answers.  Actually , we made it works ,  it looks like my colleague forgot to rename her files js in jsx . 
Now everything is fine :)
have a nice day
